Add python variant of resonant circuit (#498)
precice/tutorials: resonant-circuit/README.md

* *MATLAB* A solver using the [MATLAB bindings](https://precice.org/installation-bindings-matlab.html).
 Before running this tutorial, follow the [instructions](https://precice.org/installation-bindings-matlab.html) to correctly install the MATLAB bindings.

> * *Python* A solver using the preCICE [Python bindings](https://precice.org/installation-bindings-python.html). This solver also depends on the Python libraries `numpy` and `scipy`, which you can get from your system package manager or with `pip3 install --user <package>`.

Reviewer: Let's not suggest the pip install method here, since this is now changing and disallowed, e.g., in Ubuntu 24.04.

```suggestion
* *Python* A solver using the preCICE [Python bindings](https://precice.org/installation-bindings-python.html). This solver also depends on the Python libraries `numpy` and `scipy`.
```

Author: ```suggestion
* *Python* A solver using the preCICE [Python bindings](https://precice.org/installation-bindings-python.html). 
```
Good point. I would even go one step further: Dependencies are specified in the `requirements.txt`. If users run the solver as intended they don't have to worry about any dependencies. Let's not mention any dependencies here.

> * *Python* A solver using the preCICE [Python bindings](https://precice.org/installation-bindings-python.html). This solver also depends on the Python libraries `numpy` and `scipy`, which you can get from your system package manager or with `pip3 install --user <package>`.

Author: ```suggestion
* *Python* A solver using the preCICE [Python bindings](https://precice.org/installation-bindings-python.html). 
```
Good point. I would even go one step further: Dependencies are specified in the `requirements.txt`. If users run the solver as intended they don't have to worry about any dependencies. Let's not mention any dependencies here.

By doing that, you can now open two shells and switch into the directories `capacitor-matlab` and `coil-matlab` and execute the `run.sh` scripts.

> ### Python
> 
> This example can be executed by starting the `run.sh` file located in each of the directories `coil-python` and `capacitor-python`.
> 

Author: ```suggestion
```
No need to explicitly mention this. We have the convention that all tutorials are run this way.

## Post-processing

> The MATLAB participant for the current also records the current and voltage through time and at the end of the simulation saves a plot with the obtained curves, as well as the analytical solution.

Reviewer: How can I check what the Python participant computes?

Author: We have a `plot-solution.sh` script that uses the watchpoints.

Author: If this is not clear enough: Feel free to add some notes in the documentation.

Reviewer: This should be in this section, otherwise I need to guess that there is a script.

Also: The watchpoints seem to not be specified in the precice-config.

Author: Good point. Fixed it.

Reviewer: ![plot](https://github.com/user-attachments/assets/aba2083f-86fb-4b44-8b79-8747e23378b7)

Works, thanks!

## Post-processing

As we defined a watchpoint on the 'Capacitor' participant (see `precice-config.xml`), we can plot it with gnuplot using the script `plot-solution.sh.` You need to specify the directory of the selected solid participant as a command line argument, so that the script can pick-up the desired watchpoint file, e.g. `./plot-solution.sh capacitor-python`. The resulting graph shows the voltage and current exchanged between the two participants.

> Additionally, the MATLAB participant for the current also records the current and voltage through time and at the end of the simulation saves a plot with the obtained curves, as well as the analytical solution.

Author: ```suggestion
Additionally, the MATLAB participant `capacitor-matlab` records the current and voltage over time. At the end of the simulation it creates a plot with the computed waveforms of current and voltage, as well as the analytical solution.
```
